### DISCUSSION TITLE
not sure how to handle a general approach to changing contract details. NOOB

### DISCUSSION THREAD

**Discussion by ghost** [lbilli/Jib.jl#37] [Q&A] [answered]
Hi there

I am a julia noob coming from python and Jib.jl is teaching me ALOT about coding practices in julia, all hail ibilli. This has me stumped.

I am revisiting futures and am puzzled about how to write the call for getting the FRONT MONTH only.  I understand the call in specific terms ( when I know the contract ID) but how can I generalize it please?
```
contract = Jib.Contract("ES",...)  
Jib.reqContractDetails(ib, idx, contract)
Jib.reqMktData(ib, 1, Jib.Contract(conId=461318816, exchange="GLOBEX"), "", false)
```

So if I want to get the  contract id for the **current front month ONLY**  of ES or CL what is the process please?

thank you for any help.

**Comment by lbilli** [comment, marked as answer]
IB's [continuous future contracts](https://guides.interactivebrokers.com/tws/usersguidebook/technicalanalytics/continuous.htm) may be what you are looking for.

Using the API, following [IB docs](https://interactivebrokers.github.io/tws-api/basic_contracts.html#fut), you can use `secType="CONTFUT"` to get the lead month contract, as in:
```julia
cfut = Jib.Contract(symbol="CL", secType="CONTFUT", exchange="NYMEX")
Jib.reqContractDetails(ib, idx, cfut)

# Obtain the conId from the returned contract and then
Jib.reqMktData(ib, 1, Jib.Contract(conId=xxx, exchange="NYMEX"), "", false)
```
Continuous contracts can be used directly in `reqContractDetails()` and `reqHistoricalData()` but not in `reqMktData()`, for which a two step approach as above is required.

**Reply by ghost** [reply]
thank you so much @lbilli  I "thought" I tried this last go around and failed miserably.  I will try again next week. Thanks again for making a poor noob's life a tad easier with Jib.jl  REALLY helps me hone my "skills"
sandbagger
